src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -450,4 +471,56 @@ public List<T> scatterGather(DetachedCriteria criteria, int start, int numRows)
     protected Field getKeyField() {
         return this.keyField;
     }
+
+    @Getter
+    public static class ReadOnlyContext<T> {
+        private final int shardId;
+        private final SessionFactory sessionFactory;
+        private final boolean skipTransaction;
+        private Supplier<List<T>> reader;
+
+        public ReadOnlyContext(
+                RelationalDao<T> relationalDao,
+                int shardId,
+                SessionFactory sessionFactory,
+                DetachedCriteria criteria,
+                int first,
+                int numResults) {
+            this.shardId = shardId;
+            this.sessionFactory = sessionFactory;
+            this.reader = () -> {
+                try {
+                    return relationalDao.select(this, criteria, first, numResults);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            };
+            val skipFlag = System.getProperty("lookup.ro.skipTxn");
```

Comment:
Rename this flag to ro.skipTxn both here as well as `LookupDao`

Reply:
Done